libs/amount/amount.cpp

```diff
@@ -100,15 +90,26 @@ namespace amount {
     return *this;
   }
 
+  // to raise to power integer values
+  int Amount::ipow(int base, int exp) const {
+    int result = 1;
+    while (exp != 0) {
+      if (exp & 1) result *= base;
+      exp >>= 1;
+      base *= base;
+    }
+
+    return result;
+  }
+
   int Amount::compareTo(const Amount &other) const {
     if (precision_ == other.precision_) {
       return (value_ < other.value_) ? -1 : (value_ > other.value_) ? 1 : 0;
     }
     // when different precisions transform to have the same scale
     auto max_precision = std::max(precision_, other.precision_);
-    auto val1 = value_ * uint256_t(std::pow(10, max_precision - precision_));
-    auto val2 = other.value_ *
-                uint256_t(std::pow(10, max_precision - other.precision_));
+    auto val1 = value_ * ipow(10, max_precision - precision_);
```

Comment:
Also you can extract logic of setting proper precision to the, e.g `change_precision`, function (easy error-handling & more readable code)

Reply:
This code is not that hard. I don't think we need separate function for the single line operation

Reply:
That's not about the complexity, it is basically a separate functionality that can (and probably will) be  reused in future